chain/network/src/routing_table_actor.rs

```diff
@@ -14,22 +16,355 @@ use crate::ibf::{Ibf, IbfBox};
 use crate::ibf_peer_set::IbfPeerSet;
 #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
 use crate::ibf_set::IbfSet;
-use crate::routing::Edge;
+use crate::routing::{Edge, EdgeType, Graph, SAVE_PEERS_MAX_TIME};
 #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
 use crate::routing::{SimpleEdge, ValidIBFLevel, MIN_IBF_LEVEL};
 use crate::types::StopMsg;
 #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
-use crate::types::{PartialSync, RoutingState, RoutingVersion2};
+use crate::types::{PartialSync, PeerMessage, RoutingState, RoutingSyncV2, RoutingVersion2};
+#[cfg(feature = "delay_detector")]
+use delay_detector::DelayDetector;
+use near_primitives::utils::index_to_bytes;
+use near_store::db::DBCol::{ColComponentEdges, ColLastComponentNonce, ColPeerComponent};
+use near_store::{Store, StoreUpdate};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
 
-/// Actor that maintains routing table information.
-/// TODO (PIOTR, #4859) Finish moving routing table computation to new thread.
-#[derive(Default)]
+#[derive(Debug, Eq, PartialEq)]
+pub enum Prune {
+    Allow,
+    Force,
+    Disable,
+}
+
+/// RoutingTableActor that maintains routing table information. We currently have only one
+/// instance of this actor.
+///
+/// We store following information
+///   - list of all known edges
+///   - helper data structure for exchanging routing table
+///   - routing information (where a message should be send to reach given peer)
+///  
+/// We use store for following reasons:
+///   - store removed edges to disk
+///   - we currently don't store active edges to disk
 pub struct RoutingTableActor {
-    /// Data structures with all edges.
-    edges_info: HashMap<(PeerId, PeerId), Edge>,
+    /// Data structure with all edges.
+    pub edges_info: HashMap<(PeerId, PeerId), Edge>,
```

Comment:
and here we guarantee that first peer < second peer, right?

Reply:
Yes. I updated comment to
```
    /// Data structure with all edges. It's guaranteed that `peer.0` < `peer.1`.
```